packages/flutter_tools/lib/src/android/build_validation.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/common.dart';
+import '../build_info.dart';
+
+const String kGooglePlayVersioning = 'https://developer.android.com/studio/publish/versioning.html';
+
+/// Validates that the build mode and build number are valid for a given build.
+void validateBuild(AndroidBuildInfo androidBuildInfo) {
+  final BuildInfo buildInfo = androidBuildInfo.buildInfo;
+  if (buildInfo.mode.isPrecompiled && androidBuildInfo.targetArchs.contains(AndroidArch.x86)) {
+    throwToolExit('Cannot build ${androidBuildInfo.buildInfo.mode.name} mode for x86 ABI.');
```

Comment:
Uber nit: Link to https://flutter.dev/docs/deployment/android#what-are-the-supported-target-architectures

Reply:
Ahh, good idea!